### PR TITLE
as a end user I would like to be able to link youtube videos to my status update on gitcoin, so i can share memes and video content 

### PR DESCRIPTION
**Pull request by zoek1**
##### Description

1. parse it and show a video attachment preview in the 'post status update' module
2. when the status update is posted, the video is embedded and video can be played inline.

Home _ Gitcoin

##### Refers/Fixes

#5880

### CODE REVIEW

**Comment by zoek1** [comment]
@owocki the thumbnail match you UI criteria?

**Comment by owocki** [comment]
ui looks good;  could we easily extend this parser to work with ANY and ALL links instead of just youtube?  ie it finds the title/desc/twitter card/fb:share info and dispalys that in the same card?

i'd def throw in some more ETH for that

File: app/assets/v2/js/status.js [hunk 1]
```diff
@@ -22,6 +23,37 @@ $(document).ready(function() {
   // dropdown for usernames when @ is detected in the post
   $('#textarea').on('input', function(e) {
     const lastWord = e.target.value.split(' ').pop();
+    const youtube_re = /^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#\&\?]*).*/;
```
Comment (KiChjang) [inline app/assets/v2/js/status.js]: This looks like it only matches shortened YouTube URLs, i.e. youtu.be/dak3sl5hf32, and it only matches the beginning of the string. I found one that is more comprehensive and matches all possible combinations of YouTube's URLs:
```suggestion
    const youtube_re = /(?:https?://|//)?(?:www\.|m\.)?(?:youtu\.be/|youtube\.com/(?:embed/|v/|watch\?v=|watch\?.+&v=))([\w-]{11})(?![\w-])/;
```

**Comment by zoek1** [comment]
Sure @owocki it could be extender easily. Do you prefer in this PR or another? To avoid CORS restrictions in the browser I'll implement one endpoint to retrieve such info from the server.

File: app/assets/v2/js/status.js (hunk 1 above)
Comment (zoek1) [reply app/assets/v2/js/status.js]: Thanks, i appreciate your feedback! The screenshot shows the short and large URLs are working. I'll try your recommendations 🙂

Comment (zoek1) [reply app/assets/v2/js/status.js]: I'll substitute the regex in favor of the proposed by @KiChjang because of improving readability. :muscle:

**Comment by zoek1** [comment]
I decided to implement the extract of metadata of pages links in another PR to maintain this focused on youtube.

**Comment by zoek1** [comment]
Here is the metadata page extractor: #5889

@owocki

**Comment by zoek1** [comment]
Close in favor of #5889

**Comment by owocki** [comment]
@zoek1 you can get CORS token from the original payload look at how i do it in activity.js

**Pull request closed without merging**